app/src/main/java/com/example/myapplication/ui/LoginFragment.kt

```diff
@@ -1,16 +1,45 @@
 package com.example.myapplication.ui
 
+import android.content.Context
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
 import com.example.myapplication.databinding.FragmentLoginBinding
+import com.example.myapplication.vm.LoginViewModel
+
+class LoginFragment : Fragment() {
 
-class LoginFragment: Fragment() {
     private lateinit var binding: FragmentLoginBinding
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+
+    private lateinit var loginViewModel: LoginViewModel
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
         binding = FragmentLoginBinding.inflate(inflater, container, false)
+        loginViewModel = ViewModelProvider(this).get(LoginViewModel::class.java)
+        binding.lifecycleOwner = this
+        binding.loginViewModel = loginViewModel
+        val savedEmail = loginViewModel.sharedPreferences.getString("USERNAME", null)
```

Comment:
Instead of using the sharedPreferences, you could have a method in the viewModel and use it like this:
loginViewModel.retrieveSavedUser() : User?